src/dataServices/hybridDataService.ts

```diff
@@ -0,0 +1,314 @@
+                };
+            });
+            // clone the temporary stripped resource
+            const resourceClone = clone(resource);
+            // Resource the original JSON
+            attributesToRestore.forEach((element) => {
+                // eslint-disable-next-line no-param-reassign
+                resource[element.name] = element.data;
+            });
+            return resourceClone;
+        }
+        return resource;
+    }
+
+    private assertValidTenancyMode(tenantId?: string) {
+        if (this.enableMultiTenancy && tenantId === undefined) {
+            throw new Error('This instance has multi-tenancy enabled, but the incoming request is missing tenantId');
+        }
+        if (!this.enableMultiTenancy && tenantId !== undefined) {
+            throw new Error('This instance has multi-tenancy disabled, but the incoming request has a tenantId');
+        }
+    }
+
+    async readResource(request: ReadResourceRequest): Promise<GenericResponse> {
+        this.assertValidTenancyMode(request.tenantId);
+        const getResponse = await this.dbPersistenceService.readResource(request);
+        const replaceObjectResult = await HybridDataService.replaceStrippedResourceWithS3Version(getResponse.resource);
+        if (replaceObjectResult) {
+            return {
+                message: getResponse.message,
+                resource: replaceObjectResult,
+            };
+        }
+        return getResponse;
+    }
+
+    async vReadResource(request: vReadResourceRequest): Promise<GenericResponse> {
+        this.assertValidTenancyMode(request.tenantId);
+        const getResponse = await this.dbPersistenceService.vReadResource(request);
+        const replaceObjectResult = await HybridDataService.replaceStrippedResourceWithS3Version(getResponse.resource);
+        if (replaceObjectResult) {
+            return {
+                message: getResponse.message,
+                resource: replaceObjectResult,
+            };
+        }
+        return getResponse;
+    }
+
+    async createResource(request: CreateResourceRequest) {
+        this.assertValidTenancyMode(request.tenantId);
+        const { resourceType, resource, tenantId } = request;
+        return this.createResourceWithId(resourceType, resource, uuidv4(), tenantId);
+    }
+
+    async createResourceWithId(resourceType: string, resource: any, resourceId: string, tenantId?: string) {
+        if (this.shallStoreOnObjectStorage(resourceType)) {
+            const fileName = this.getPathName(resourceId, uuidv4(), resourceType, tenantId);
+
+            const resourceClone = clone(resource);
+            resourceClone.id = resourceId;
+            const base64Data = encode(JSON.stringify(resourceClone));
+
+            // remove the main payload.
+            const strippedResource = this.stripPayloadFromResource(resourceType, resourceClone);
+
+            // link the s3 key to the stripped resource
+            if (!strippedResource.meta) {
```

Comment:
Suggestion: wrap up this statement in a function. Less code duplication + with proper naming no need to carry comments